core/thread.c

```diff
@@ -124,6 +124,19 @@ void thread_add_to_list(list_node_t *list, thread_t *thread)
     list->next = new_node;
 }
 
+void thread_remove_from_list(list_node_t *list, thread_t *thread)
```

Comment:
I didn't consider it because: the list is not a circular list and because when added is using a `thread_add_to_list` which is in the thread module (I assume that this is done this way to lower inter-dependencies).